lib/moss_theme/src/index.ts

```diff
@@ -1,109 +1,83 @@
-// Theme used for JSON parsing
-export interface Theme {
-  name: string;
-  type: string;
-  default: boolean;
-  colors: Colors;
-}
-
-export interface Colors {
-  primary?: string;
-  sideBarBackground?: string;
-  toolBarBackground?: string;
-  pageBackground?: string;
-  statusBarBackground?: string;
-  windowsCloseButtonBackground?: string;
-  windowControlsLinuxBackground?: string;
-  windowControlsLinuxText?: string;
-  windowControlsLinuxHoverBackground?: string;
-  windowControlsLinuxActiveBackground?: string;
-}
+import { Theme, Colors } from "@repo/moss-models";
 
+// Mapping between Theme JSON and TypeScript types
 const typeMap: any = {
   Theme: createPropsWithAdditional(
-    [
-      { json: "name", js: "name", typ: "" },
-      { json: "type", js: "type", typ: "" },
-      { json: "default", js: "default", typ: true },
-      { json: "colors", js: "colors", typ: createReference("Colors") },
-    ],
+    Object.keys(new Theme("", "", false, new Colors())).map((key) => ({
+      json: key,
+      js: key,
+      typ: key === "colors" ? createReference("Colors") : key === "isDefault" ? true : "",
+    })),
     false
   ),
   Colors: createPropsWithAdditional(
-    [
-      { json: "primary", js: "primary", typ: createUnionMembers(undefined, "") },
-      { json: "sideBar.background", js: "sideBarBackground", typ: createUnionMembers(undefined, "") },
-      { json: "toolBar.background", js: "toolBarBackground", typ: createUnionMembers(undefined, "") },
-      { json: "page.background", js: "pageBackground", typ: createUnionMembers(undefined, "") },
-      { json: "statusBar.background", js: "statusBarBackground", typ: createUnionMembers(undefined, "") },
-      {
-        json: "windowsCloseButton.background",
-        js: "windowsCloseButtonBackground",
-        typ: createUnionMembers(undefined, ""),
-      },
-      {
-        json: "windowControlsLinux.background",
-        js: "windowControlsLinuxBackground",
-        typ: createUnionMembers(undefined, ""),
-      },
-      { json: "windowControlsLinux.text", js: "windowControlsLinuxText", typ: createUnionMembers(undefined, "") },
-      {
-        json: "windowControlsLinux.hoverBackground",
-        js: "windowControlsLinuxHoverBackground",
-        typ: createUnionMembers(undefined, ""),
-      },
-      {
-        json: "windowControlsLinux.activeBackground",
-        js: "windowControlsLinuxActiveBackground",
+    Object.keys(new Colors()).map((key) => {
+      return {
+        json: key,
+        js: key,
```

Comment:
If I understand everything correctly, then here we can reduce code duplication a little and improve typing a little

Something like: 

```typescript
interface PropsWithAdditional {
  json: string;
  js: string;
  typ: any;
}

function createPropsFromObject(obj: object, keyTypeCallback: (key: string) => any): PropsWithAdditional[] {
  return Object.keys(obj).map((key) => ({
    json: key,
    js: key,
    typ: keyTypeCallback(key),
  }));
}

function keyTypeForTheme(key: string): any {
  if (key === "colors") {
    return createReference("Colors");
  } else if (key === "isDefault") {
    return true;
  }
  return "";
}

function keyTypeForColors(): any {
  return createUnionMembers(undefined, "");
}

const typeMap: Record<string, PropsWithAdditional[]> = {
  Theme: createPropsWithAdditional(createPropsFromObject(new Theme("", "", false, new Colors()), keyTypeForTheme), false),
  Colors: createPropsWithAdditional(createPropsFromObject(new Colors(), keyTypeForColors), false),
};
```